docs/src/Fieldset.doc.js

```diff
@@ -0,0 +1,222 @@
+// @flow strict
+import type { Node } from 'react';
+import { Fieldset } from 'gestalt';
+import PropTable from './components/PropTable.js';
+import PageHeader from './components/PageHeader.js';
+import MainSection from './components/MainSection.js';
+
+const cards: Array<Node> = [];
+const card = (c) => cards.push(c);
+
+card(
+  <PageHeader
+    name="Fieldset"
+    description="Creates an accessible fieldset and legend for a group of related form items"
+    defaultCode={`
+      function RadioButtonExample() {
+        const [favorite, setFavorite] = React.useState(undefined);
+
+        return (
+          <Fieldset legend="What is your favorite pet?">
+            <Flex direction="column" gap={2}>
+              <RadioButton
+                checked={favorite === 'dogs'}
+                id="favoriteDog"
+                label="Dogs"
+                name="favorite"
+                onChange={() => setFavorite( 'dogs' )}
+                value="dogs"
+              />
+              <RadioButton
+                checked={favorite === 'cats'}
+                id="favoriteCat"
+                label="Cats"
+                name="favorite"
+                onChange={() => setFavorite( 'cats' )}
+                value="cats"
+              />
+              <RadioButton
+                checked={favorite === 'plants'}
+                id="favoritePlants"
+                label="Plants"
+                name="favorite"
+                onChange={() => setFavorite( 'plants' )}
+                value="plants"
+              />
+            </Flex>
+          </Fieldset>
+
+        );
+      }
+`}
+  />,
+);
+
+card(
+  <PropTable
+    Component={Fieldset}
+    props={[
+      {
+        name: 'legend',
+        type: 'string',
+        required: true,
+        description:
+          "Content for this fieldset's legend that clearly and concisely describes the question being asked.",
+      },
+      {
+        name: 'legendDisplay',
+        type: "'visible' | 'hidden'",
+        defaultValue: 'visible',
+        description:
+          'Whether the legend should be visible or not. If `hidden`, the legend is still available for screen reader users, but does not appear visually.',
+      },
+      {
+        name: 'children',
+        type: 'React.Node',
+        required: true,
+        description: `The content of Fieldset, typically [RadioButtons](/RadioButton), [Checkboxes](/Checkbox) or [TextFields](/TextField).`,
+      },
+    ]}
+  />,
+);
+
+card(
+  <MainSection name="Accessibility">
+    <MainSection.Subsection
+      title="When to use Fieldset"
```

Comment:
Could we add a good example for Textfield?  

I thought Profile Details was  legend! So i'd be great to show both cases, when Textfield are actually grouped.

![image](https://user-images.githubusercontent.com/10593890/117051069-22b8f600-ace4-11eb-9581-8cbfd4409ef6.png)